internal/cmd/verify-name-conflicts.go

```diff
@@ -0,0 +1,191 @@
+			return err
+		}
+
+		for _, release := range releases {
+			if !strings.Contains(githubObj.IgnoreVersions, release.TagName) {
+				for _, asset := range release.Assets {
+					if asset.Name == contract.Filename {
+						filePath := tempDirPath + "/" + githubObj.Type + "-" + release.TagName + "-" + githubObj.Name + ".yaml"
+						err := downloadAndParseFile(asset.BrowserDownloadURL, filePath, githubObj.Type, kindSourceMap)
+						if err != nil {
+							return err
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// downloadAndParseFile function calls the downloadFile & parseFile functions which downloads the various catalog.yaml files of each release & then parses them
+func downloadAndParseFile(url, filepath, kind string, unique map[string]map[string][]ResourceInfo) error {
+	if err := downloadFile(url, filepath); err != nil {
+		return err
+	}
+
+	if err := parseFile(filepath, kind, unique, url); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// downloadFile function downloads the file mentioned in the url & stores it in the mentioned filepath
+func downloadFile(url, filepath string) error {
+	file, err := os.OpenFile(filepath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	res, err := MakeGetRequest(url)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return fmt.Errorf("server returned non-200 status code: %v", res.Status)
+	}
+
+	if _, err = io.Copy(file, res.Body); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// parseResources function parses the resources & checks for uniqueness
+func parseResources(resources []*contract.TektonResource, unique map[string][]ResourceInfo, source string, kind string) error {
+	for _, res := range resources {
+		name := res.Name
+		version := res.Version
+		_, exists := unique[name]
+
+		if exists {
+			currResources := unique[name]
+			//Checks whether the sources are different
+			if currResources[0].Source != source {
+				return fmt.Errorf("2 resources of kind '%s', have same name '%s', from different sources, \nsource1: %s\nsource2: %s", kind, name, currResources[0].Source, source)
+			}
+			//Checks whether the versions are same or not, if its from same source
+			for _, currResource := range currResources {
+				if currResource.Version == version {
+					return fmt.Errorf("2 resources of kind '%s', have same name '%s', from same source '%s'", kind, name, source)
+				}
+			}
+			//If none of the above then the resource is still unique so append
+			unique[name] = append(unique[name], ResourceInfo{Version: version, Source: source})
+		} else {
+			//If no name conflict then resource is unique so append
+			unique[name] = []ResourceInfo{{Version: version, Source: source}}
+		}
+	}
+	return nil
+}
+
+// parseFile function parses the file present in the path & then calls the parseResources function to check for uniqueness
```

Comment:
Just a suggestion: The comment should be very generic so that it is easily understandable without needing to refer to the function definition

here in the comment 
```
parseFile function parses the file present in the path
```  
its not that cleat when you say file present in the path 

Instead can you mention something like below

// parseFile processes the YAML file of a Task or Pipeline to verify the uniqueness of names